### ISSUE TITLE
[FIRRTL] Negative Signed Literal error

### ISSUE THREAD

**Issue by drom** [llvm/circt#347]
The following FIRRTL program

```python
circuit top_mod :
  module top_mod :
    output tmp12: SInt<4>
    tmp12 <= SInt<4>(-4)
```
Compiled with `firtool --lower-to-rtl` produces this error:

```python
top_mod.fir:4:22: error: expected '.' in floating point literal
    tmp12 <= SInt<4>(-4)
                     ^
```

**Comment by seldridge** [comment]
I ran into this a while back with negative external module parameters: https://github.com/llvm/circt/issues/183

We need to both fix the parsing and come up with how to represent the literals.

**Comment by lattner** [comment]
Fixed in 362a282 thank you again @drom!
